app/registries/foreman/plugin.rb

```diff
@@ -87,10 +109,13 @@ def unregister(plugin_id)
         @registered_plugins.delete(plugin_id)
       end
 
-      # Clears the registered plugins hash
+      # Clears the registered plugins hash and registries
       # It doesn't unload installed plugins
-      def clear
-        @registered_plugins = {}
+      # You can provide registered plugins and registries hashes to use instead of clear hashes
+      def clear(plugins = {}, registries = {})
```

Comment:
I wonder if we should raise an exception if Rails is in production mode when plugins and registries are not empty. IMHO that's only really for tests.

Btw, with tests you can call private methods. Perhaps that would be a better way. A good public API should be hard to abuse.